### PR TITLE
Decompile IAs "SlidyIce" and "SlowingSnow"

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/global.h [hunk 1]
```diff
@@ -87,6 +87,8 @@
 // Converts a Q16.16 fixed-point format number to a regular integer
 #define Q_16_16_TO_INT(n) ((int)((n) >> 0x10))
 
+#define Q_24_8_MULTIPLY(intVal, floatVal) Q_24_8_TO_INT((intVal)*Q_24_8(floatVal))
```
Comment (freshollie) [inline include/global.h:90]: Idk if you need to make this a macro, feels like this isn't something they would have written

Comment (JaceCear) [reply include/global.h:90]: You think so?
I feel like it's more likely they did it, than writing out the contents of this macro every time they multiply like that.

File: src/interactable_ice_paradise_slowing_snow.c [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    snow->ia = ia;
+    snow->spriteX = ia->x;
+    snow->spriteY = spriteY;
+
+    snow->posX = SpriteGetScreenPos(ia->x, spriteRegionX);
+    snow->posY = SpriteGetScreenPos(ia->y, spriteRegionY);
+    SET_SPRITE_INITIALIZED(ia);
+}
+
+bool32 PlayerTouchesSnow(Sprite_SlowingSnow *snow)
```
Comment (freshollie) [inline src/interactable_ice_paradise_slowing_snow.c:62]: Minor point, prefer `PlayerTouchingSnow`, `Touches` doesn't provide a time context, and thus questions could be asked "Does this function return true if the player ever touches snow?"

Comment (JaceCear) [reply src/interactable_ice_paradise_slowing_snow.c:62]: I mean... it's not like there's tons of abstractions, the procudure is literally right next to it. :D
But will try to keep it in mind in the future

File: src/interactable_toggle_player_layer.c [hunk 3]
```diff
@@ -9,19 +9,26 @@
 #include "constants/interactables.h"
 #include "constants/move_states.h"
 
+typedef struct {
```
Comment (freshollie) [inline src/interactable_toggle_player_layer.c:12]: Yes much cleaner

File: src/interactable_ice_paradise_slowing_snow.c (hunk 2 above)
Comment (JaceCear) [reply src/interactable_ice_paradise_slowing_snow.c:62]: But I def should make it static

File: src/interactable_slidy_ice.c [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    /* 0x04 */ s8 offsetY;
+    /* 0x05 */ u8 width;
+    /* 0x06 */ u8 height;
+} Interactable_SlidyIce PACKED;
+
+typedef struct {
+    /* 0x00 */ SpriteBase base;
+} Sprite_SlidyIce;
+
+#if 1
```
Comment (freshollie) [inline src/interactable_slidy_ice.c:26]: Can remove these

File: include/global.h (hunk 1 above)
Comment (freshollie) [reply include/global.h:90]: IMO it's a tradeoff between readability, transparency, and shared code. In this case:

- The macro makes the function more readable
- The macro makes it less transparent what's going on
- The macro only removes a few characters of code

**Review by freshollie — APPROVED** [review]

**Pull request merged**